_posts/2022-12-21-easily-verify-your-rust-in-ci-with-kani.md

```diff
@@ -0,0 +1,80 @@
+For example, on our own Kani repo, a CI status report on a PR might look like this:
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/kani-ci-checks.png" alt="Kani CI Checks" />
+
+Similarly, **Continuous Verification** enables developers to prevent regressions before they happen.
+Kani users have asked us for an easy way to integrate continuous verification into their existing development workflows.
+To this end, we are excited to announce the [Kani Rust Verifier Action](https://github.com/marketplace/actions/kani-rust-verifier) on the GitHub Marketplace.
+
+## How to use the Kani GitHub action
+
+If you have a [Rust Cargo](https://doc.rust-lang.org/cargo/) project in a GitHub repo, using the Kani CI action is as simple as adding it to your `ci.yaml` file.
+
+To get the latest version, visit the [kani verifier action](https://github.com/marketplace/actions/kani-rust-verifier) on the marketplace, and click on the green “Use Latest Version” button, which will give you a yaml snippet you can paste into your `.yml` file.
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/kani-verifier-action.png" alt="Kani Verifier Action on GitHub Marketplace" />
+
+For example, for Kani 0.16, your `kani-ci.yml` file might look like this:
+
+```yaml
+name: Kani CI
+on:
+  pull_request:
+  push:
+jobs:
+  run-kani:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: 'Checkout your code.'
+        uses: actions/checkout@v3
+
+# You can get the latest version from
+# https://github.com/marketplace/actions/kani-rust-verifier
+      - name: 'Run Kani on your code.'
+        uses: model-checking/kani-github-action@v0.16
+```
+
+For more advanced use cases, we provide facilities to override the working directory, as well as to configure the Kani command itself.
+For example, the [s2n-quic](https://github.com/aws/s2n-quic) project uses [the following CI configuration](https://github.com/aws/s2n-quic/blob/main/.github/workflows/ci.yml#L613), which overrides the working directory, and enables the `--tests` option of `cargo kani`.
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/s2n-quic-using-kani-action.png" alt="How s2n quic integrates the Kani action" />
+
+For full details, consult [the documentation](https://model-checking.github.io/kani/install-github-ci.html).
+
+## When to run the Kani action
+
+One question we have received from our users is “When should we run the Kani action?”  Is it better to run it on every PR, once on a PR right before merge, or nightly?  The answer is an emphatic “it depends!”.
+The great thing about GitHub Actions are that you can use them in the way that best fits your workflow.
+Our experience is that the sooner verification runs, the better: it’s easier to debug and fix issues when they’re found right away, rather than after they’re merged.
+Our experience is that many Kani proofs can complete quickly on standard CI hardware.
```

Comment:
It's a little repetitive to begin two sentences in a row with "Our experience is that". Maybe the second sentence can be simplified to "Many Kani proofs complete quickly on standard CI hardware."

Reply:
Fixed